core/app/models/spree/return_authorization.rb

```diff
@@ -61,6 +62,10 @@ def can_cancel_return_items?
 
     private
 
+      def authorize_return
+        order.authorize_return! unless order.nil?
```

Comment:
@mtomov - you will note Spree already has an ```OrderPopulator``` object, which is a very similar kind of concept and could be argued to be an implementation of DCI (if you think of DCI as a broad pattern, Coplien actually defines it a little more narrowly and ```OrderPopulator``` might not fit his definition)